dpypelines/pipeline/generic_file_ingress_v1.py

```diff
@@ -0,0 +1,184 @@
+        de_notifier.failure()
+        raise err
+
+    # Create email client from env var
+    try:
+        email_client = get_email_client()
+        logger.info("Created email client", data={"email_client": email_client})
+    except Exception as err:
+        logger.error("Error occurred when creating email client", err)
+        de_notifier.failure()
+        raise err
+
+    # Get Upload Service URL from environment variable
+    try:
+        upload_url = os.environ.get("UPLOAD_SERVICE_URL", None)
+        assert (
+            upload_url is not None
+        ), "UPLOAD_SERVICE_URL environment variable not set."
+        logger.info("Got Upload Service URL", data={"upload_url": upload_url})
+    except Exception as err:
+        logger.error("Error occurred when getting Upload Service URL", err)
+        de_notifier.failure()
+        raise err
+
+    # Attempt to access the local data store
+    try:
+        local_store = LocalDirectoryStore(files_dir)
+        files_in_directory = local_store.get_file_names()
+        logger.info(
+            "Local data store created",
+            data={
+                "local_store_dir": files_dir,
+                "files_in_directory": files_in_directory,
+            },
+        )
+    except Exception as err:
+        logger.error(
+            "Error occurred when creating local data store from files directory",
+            err,
+            data={"local_store_dir": files_dir},
+        )
+        de_notifier.failure()
+        raise err
+
+    # Extract the patterns for required files from the pipeline configuration
+    try:
+        required_file_patterns = get_required_files_patterns(pipeline_config)
+        logger.info(
+            "Required file patterns retrieved from pipeline config",
+            data={"required_file_patterns": required_file_patterns},
+        )
+    except Exception as err:
+        files_in_directory = local_store.get_file_names()
+        logger.error(
+            "Error occurred when getting required file patterns",
+            err,
+            data={
+                "files_in_directory": files_in_directory,
+                "pipeline_config": pipeline_config,
+            },
+        )
+        de_notifier.failure()
+        raise err
+
+    # Check for the existence of each required file
+    for required_file in required_file_patterns:
+        try:
+            if not local_store.has_lone_file_matching(required_file):
+                email_content = file_not_found_email(required_file)
+                email_client.send(
+                    submitter_email, email_content.subject, email_content.message
+                )
+                logger.info(
+                    "Email sent to submitter about missing required file",
+                    data={
+                        "submitter_email": submitter_email,
+                        "required_file": required_file,
+                    },
+                )
+                de_notifier.failure()
```

Comment:
we should raise an error here. We've found out that a required file is missing and notified of a failure but we're not rising an error so the pipeline is just gonna keep going.

**After** the zde_notifier.failure()` can you add something like

```
raise FileNotFoundError(f"Could not find file matching pattern {required_file}")
```
Again its wrong in both pipelines so you'll need to correct in both sorry.

Reply:
Done